openhands/runtime/impl/remote/remote_runtime.py

```diff
@@ -277,6 +278,24 @@ def _wait_until_alive(self):
         assert runtime_data['runtime_id'] == self.runtime_id
         assert 'pod_status' in runtime_data
         pod_status = runtime_data['pod_status']
+
+        # FIXME: We should fix it at the backend of /start endpoint, make sure
+        # the pod is created before returning the response.
+        # Retry a period of time to give the cluster time to start the pod
+        n_attempts = 0
+        while pod_status == 'Not Found' and n_attempts < 10:
+            time.sleep(5)
+            runtime_info_response = self._send_request(
+                'GET',
+                f'{self.config.sandbox.remote_runtime_api_url}/runtime/{self.runtime_id}',
+            )
+            runtime_data = runtime_info_response.json()
+            assert 'runtime_id' in runtime_data
+            assert runtime_data['runtime_id'] == self.runtime_id
+            assert 'pod_status' in runtime_data
+            pod_status = runtime_data['pod_status']
+            n_attempts += 1
+
```

Comment:
Let's not do our own retry logic. Instead, you can just `raise RuntimeNotReadyException` below to trigger tenacity

Maybe we should move this into its own PR too

Reply:
Reverted!

Reply:
done - and move these stuff to https://github.com/All-Hands-AI/OpenHands/pull/4825